chore: update dependencies

Tigge opened the pull request:
Update dependencies to the latest version. This includes an update of
`eslint-config-typescript-shareable` that makes a few import/* rules
function as intended again.

Most of the changes in this commit are related to that update. Most are
simple import changes from `.` to `./Component`. On a few occasions, some
parts were refactored out.

anius commented:
- Please update `react-hooks-shareable`  to `1.13.0`
- Please remove `I` from the files you changes as we get breaking changes anyway from the previous commit 😎

Tigge commented:
> * Please update `react-hooks-shareable`  to `1.13.0`
> * Please remove `I` from the files you changes as we get breaking changes anyway from the previous commit sunglasses

Updated to latest deps. Added one additional commit that normalizes the interface names.

lekoaf reviewed (approved):
> +4,753 −7,878

I'll take your word for it... :see_no_evil:
